tsdb/db.go

```diff
@@ -839,21 +884,22 @@ func (db *DB) compactHead(head *RangeHead) (err error) {
 			var merr tsdb_errors.MultiError
 			merr.Add(errors.Wrap(err, "reload blocks"))
 			merr.Add(errors.Wrapf(errRemoveAll, "delete persisted head block after failed db reload:%s", uid))
-			return merr.Err()
+			return walTruncationTime, merr.Err()
 		}
-		return errors.Wrap(err, "reload blocks")
+		return walTruncationTime, errors.Wrap(err, "reload blocks")
 	}
+	walTruncationTime = maxt
 	if (uid == ulid.ULID{}) {
-		// Compaction resulted in an empty block.
-		// Head truncating during db.reload() depends on the persisted blocks and
 		// in this case no new block will be persisted so manually truncate the head.
-		if err = db.head.Truncate(maxt); err != nil {
-			return errors.Wrap(err, "head truncate failed (in compact)")
+		// Head truncating during db.reload() depends on the persisted blocks and
+		// Compaction resulted in an empty block.
+		if err = db.head.TruncateInMemory(maxt); err != nil {
+			return walTruncationTime, errors.Wrap(err, "head truncate failed (in compact)")
```

Comment:
This will return `walTruncationTime = maxt` but, unless I'm missing something, the returned `walTruncationTime` is never used if there's an error. I'm wondering if the could would be easier to follow if you always `return math.MinInt64` and at the end of this function you `return maxt, nil`